File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        stack = []
        node = root
        valLeft = k
        while node:
            stack.append(node)
            node = node.left
        while k > 1:
            item = stack.pop()
            k -= 1
            if item.right:
                stack.append(item.right)
        return stack.pop().val
            

        

File: Data Structures & Algorithms/balanced-binary-tree/submission-3.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {boolean}
     */
    isBalanced(root) {
        let isCorrect = true;
        function bal(node) {

            if (node === null) {
                return 0;
            }
            const left = bal(node.left);
            const right = bal(node.right);
            if (Math.abs(left - right) > 1) {
                isCorrect = false;
            }
            return Math.max(left, right) + 1
        }
        bal(root);
        return isCorrect;
    }


}
